src/imageops/colorops.rs

```diff
@@ -72,16 +70,10 @@ where
 /// Invert each pixel within the supplied image.
 /// This function operates in place.
 pub fn invert<I: GenericImage>(image: &mut I) {
-    // TODO why is this not just `for pixel in image.pixels_mut() { pixel.invert(); }`
-    let (width, height) = image.dimensions();
-
-    for y in 0..height {
-        for x in 0..width {
-            let mut p = image.get_pixel(x, y);
-            p.invert();
-
-            image.put_pixel(x, y, p);
-        }
+    let mut image = RefCell::new(image);
+    for (x, y, mut pixel) in image.borrow().pixels() {
+        pixel.invert();
+        image.borrow_mut().put_pixel(x, y, pixel);
```

Comment:
Do any of the tests execute this line? I'd expect it to panic at runtime because the borrow at the top of the loop is still in scope.

Reply:
No none of the tests were currently running this code. I did some tests function and it didn't work at runtime.  
I'm trying to figure out a solution.

Reply:
It works by using `addr_of_mut!` and some `unsafe` code. I have added a few tests to make sure and it works at both compile-time and runtime.
But I'm not sure if its the best solutions.